### PR TITLE
build: upgrade dependency versions

### LINKED ISSUE #1012: Dependency Review and Upgrade
I'd like to request, and help preform, a review of the dependencies and upgrade them as necessary. I'd like to try to use Spinal at my office and they would prefer more up-to-date libraries than what are being used now.

### PR DESCRIPTION
**Pull request by kleinai**
Closes #1012

Flat upgrading all the dependencies seems to have worked well with only one incompatibility.

# Context, Motivation & Description

Upgrade dependencies to keep current.

# Impact on code generation

None expected.

### CODE REVIEW

**Review by numero-744 — APPROVED** [review]

**Comment by numero-744** [comment]
Thanks for the PR!

It seems GitHub has network issues (see https://github.com/SpinalHDL/SpinalHDL/pull/1015/checks), I'm waiting for them to be fixed before I merge this PR (to avoid failed workflow on `dev`).

**Comment by Readon** [comment]
This might affect the projects based on this library like NaxRiscv, VexRiscv and a bundle of other works, which would need an update on dependency also.
I still suggest to wait most of other projects be tested first.

**Pull request merged**
